### DISCUSSION TITLE
Is there API to get type of a context?

### DISCUSSION THREAD

**Discussion by zhaoyao91** [solidjs/solid#457] [Q&A] [answered]
say, we have a context `Context<MyValue>`, how to extract type `MyValue`?

**Comment by amoutonbrady** [comment]
I may have one or two examples to point to, but could you first provide a more complete example of what you are trying to achieve and can't do?

**Reply by zhaoyao91** [reply]
I am making a library which export  a predefined context to users. Currently I have to define another `XXContenxtValue` and export it with the context, but if there is standard way to extract the value type of the context, thing could be cleaner.

**Comment by danielpza** [comment, marked as answer]
Perhaps this is what you want https://www.typescriptlang.org/docs/handbook/2/conditional-types.html#inferring-within-conditional-types

```ts
type ContextType<T extends Context<any>> = T extends Context<infer R> ? R : never;

type MyContext = Context<number>

const value: ContextType<MyContext> // number
```

**Reply by zhaoyao91** [reply]
Thanks for this clever method. (p.s. solid even not export `Context` type)

**Reply by ryansolid** [reply]
I thought I had addressed most of these in: https://github.com/solidjs/solid/commit/f870851c931d68a44a25ff8ac68d895cc1f3333c. But it looks like I still missed Context. Let me know if you come across any others.
